compiler/z/codegen/OMRSnippet.cpp

```diff
@@ -149,7 +149,8 @@ OMR::Z::Snippet::generatePICBinary(TR::CodeGenerator * cg, uint8_t * cursor, TR:
          }
 #endif
 
-      TR_ASSERT(CHECK_32BIT_TRAMPOLINE_RANGE(destAddr, cursor),  "Helper Call is not reachable.");
+      TR_ASSERT(TR::Compiler->target.cpu.isTargetWithinBranchRelativeRILRange(destAddr, (intptrj_t)cursor),
+                "Helper Call is not reachable.");
```

Comment:
Let's make this a fatal assert. I'd rather have a compile time assert failure than unpredictable runtime crash.

Reply:
Similarly elsewhere in Z codegen where asserts were touched.

Reply:
Agreed.